src/app/qgsmapthemes.cpp

```diff
@@ -156,11 +156,17 @@ void QgsMapThemes::applyState( const QString &presetName )
 
 void QgsMapThemes::removeCurrentPreset()
 {
-  Q_FOREACH ( QAction *a, mMenuPresetActions )
+  for ( QAction *actionPreset : qgis::as_const( mMenuPresetActions ) )
   {
-    if ( a->isChecked() )
+    if ( actionPreset->isChecked() )
     {
-      QgsProject::instance()->mapThemeCollection()->removeMapTheme( a->text() );
+      int res = QMessageBox::question( mMenu, tr( "Remove Theme" ),
+                                       trUtf8( "Are you sure you want to remove the existing theme “%1”?" ).arg( actionPreset->text() ),
+                                       QMessageBox::Yes | QMessageBox::No, QMessageBox::No );
+      if ( res != QMessageBox::Yes )
+        return;
```

Comment:
Intermittent returns are a dangerous thing. For example, if one introduces a `mapThemesChanged` signal and emits it, it's too easy to not see this return here.
Instead it's safer to use

```
if ( res == QMessageBox::Yes )
  QgsProject::instance()->mapThemeCollection()->removeMapTheme( actionPreset->text() ); 
```